src/components/Download.tsx

```diff
@@ -1,10 +1,232 @@
+import { useState, useMemo } from 'react';
+import {
+  Button,
+  Alert,
+  Collapse,
+  Typography,
+  List,
+  ListItem,
+  Switch,
+  FormControlLabel,
+  Link,
+} from '@mui/material';
+import Ajv from 'ajv';
 import NavigationButton from './NavigationButton';
+import DataDictionaryPreview from './DataDictionaryPreview';
+import useDataStore from '../stores/data';
+import schema from '../assets/neurobagel_data_dictionary.schema.json';
+import { DataDictionary } from '../utils/types';
+import emoji from '../assets/download-emoji.png';
 
 function Download() {
+  const [dictionaryCollapsed, setDictionaryCollapsed] = useState(false);
+  const [forceAllowDownload, setForceAllowDownload] = useState(false);
+
+  const uploadedDataTableFileName = useDataStore((state) => state.uploadedDataTableFileName);
+
+  const columns = useDataStore((state) => state.columns);
+
+  const dataDictionary = useMemo(
+    () =>
+      Object.entries(columns).reduce((acc, [_, column]) => {
+        if (column.header) {
+          acc[column.header] = {
+            Description: column.description || '',
+          };
+        }
+        return acc;
+      }, {} as DataDictionary),
+    [columns]
+  );
+
+  const { isValid: schemaValid, errors: schemaErrors } = useMemo(() => {
+    const ajv = new Ajv();
+    const validate = ajv.compile(schema);
+    const isValid = validate(dataDictionary);
+
+    if (!isValid) {
+      const errors = validate.errors?.map((error) => error.instancePath.slice(1)) || [];
+      return { isValid: false, errors };
+    }
+
+    return { isValid: true, errors: [] };
+  }, [dataDictionary]);
+
+  const handleDownload = () => {
+    const blob = new Blob([JSON.stringify(dataDictionary, null, 2)], {
+      type: 'application/json;charset=utf-8',
+    });
+
+    const url = URL.createObjectURL(blob);
+    const a = document.createElement('a');
+    a.href = url;
+    a.download = `${uploadedDataTableFileName}_annotated.json`;
```

Comment:
you should strip the file ending up `uploadedDataTableFileName` first - and have a test to ensure that works.

Reply:
Ah good catch!